vignettes/04-synthesis-data.Rmd

```diff
@@ -1 +1,216 @@
+trait_weather_df = trait_weather_df %>% 
+  select(day, cultivar, mean, gdd_cum) %>% 
+  na.omit()
+```
+
+## Plot and model relationship between GDD and canopy cover for each cultivar
+
+We are interested in how growing degree days affects canopy cover. 
+To investigate this, we are going to model and plot their relationship. 
+We want to know the relationship for each cultivar, so we'll start of by determining the parameters of the model for one of the cultivars in our dataset. 
+We are using a logistic growth model here because it is appropriate for the shape of the GDD-cover relationship. 
+
+```{r model_get_parameters}
+single_cultivar <- trait_weather_df %>% 
+  filter(cultivar == "PI656026")
+
+cap     <- 120
+initial <- 25
+mean    <- single_cultivar$mean[57]
+gdd_cum <- single_cultivar$gdd_cum[57]
+rate    <- ((log((cap/mean) - 1)) - initial)/gdd_cum
+
+model_single_cultivar <- nls(mean ~ cap / (1 + exp(initial + rate * gdd_cum)), 
+                             start = list(cap = 120, initial = 25, rate = rate),
+                             data = single_cultivar, trace = TRUE)
+
+single_cultivar <- single_cultivar %>% 
+  mutate(mean_predict = coef(model_single_cultivar)[1] / (1 + exp(coef(model_single_cultivar)[2] + coef(model_single_cultivar)[3] * gdd_cum)))
+```
+
+We then use the parameters from a single cultivar to run a model for each of the rest of the cultivars. 
+These results are used to plot the model predictions, which are shown as an orange line. 
+
+```{r model_all_cultivars}
+all_cultivars <- c(day = as.double(), cultivar = as.character(), mean = as.numeric(), 
+                   gdd_cum = as.numeric(), mean_predict = as.numeric())
+for(each_cultivar in unique(trait_weather_df$cultivar)){
+  each_cultivar_df <- filter(trait_weather_df, cultivar == each_cultivar)
+  each_cultivar_model <- nls(mean ~ cap / (1 + exp(initial + rate * gdd_cum)), 
+                            start = list(cap = 120, initial = 25, rate = rate),
+                            data = each_cultivar_df)
+  each_cultivar_df <- each_cultivar_df %>% 
+      mutate(mean_predict = coef(each_cultivar_model)[1] / (1 + exp(coef(each_cultivar_model)[2] + coef(each_cultivar_model)[3] * gdd_cum)))
+  all_cultivars <- rbind(each_cultivar_df, all_cultivars)
+}
+
+ggplot(all_cultivars) +
+  geom_point(aes(x = gdd_cum, y = mean)) +
+  geom_line(aes(x = gdd_cum, y = mean_predict), color = "orange") +
+  facet_wrap(~cultivar, scales = "free_y") +
+  labs(x = "Cumulative growing degree days", y = "Canopy Height")
+```
+
+## Create histogram of growth rate for all cultivars
+
+The last thing that we are going to do is assess the difference in this relationship among the cultivars. 
+We are going to use the inflection point from the logistic growth model, which indicates when canopy cover stops increasing as quickly with increasingly more warm days. 
+The `inflection` R package is used for the calculation. 
```

Comment:
please use analytical solution https://stats.stackexchange.com/a/39262 instead of this package.

Reply:
See 9c16e81